Habitica/src/main/java/com/habitrpg/android/habitica/helpers/LanguageHelper.java

```diff
@@ -0,0 +1,120 @@
+                break;
+            case "bg":
+                locale = new Locale("bg");
+                languageCode = "bg";
+                languageAvailable = true;
+                break;
+            case "de":
+                locale = new Locale("de");
+                languageCode = "de";
+                languageAvailable = true;
+                break;
+            case "en-rGB":
+                locale = new Locale("en", "GB");
+                languageCode = "en_GB";
+                languageAvailable = true;
+                break;
+            case "es":
+                locale = new Locale("es");
+                languageCode = "es";
+                languageAvailable = true;
+                break;
+            case "fr":
+                locale = new Locale("fr");
+                languageCode = "fr";
+                languageAvailable = true;
+                break;
+            case "iw":
+                locale = new Locale("iw");
+                languageCode = "he";
+                languageAvailable = true;
+                break;
+            case "hr":
+                locale = new Locale("hr", "HR");
+                languageCode = "hu";
+                languageAvailable = true;
+                break;
+            case "in":
+                locale = new Locale("in");
+                languageCode = "id";
+                languageAvailable = true;
+                break;
+            case "it":
+                locale = new Locale("it");
+                languageCode = "it";
+                languageAvailable = true;
+                break;
+            case "ja":
+                locale = new Locale("ja");
+                languageCode = "ja";
+                languageAvailable = true;
+                break;
+            case "nl":
+                locale = new Locale("nl");
+                languageCode = "nl";
+                languageAvailable = true;
+                break;
+            case "pl":
+                locale = new Locale("pl");
+                languageCode = "pl";
+                languageAvailable = true;
+                break;
+            case "pt":
+                locale = new Locale("pt","PT");
+                languageCode = "pt";
+                languageAvailable = true;
+                break;
+            case "pt-rBR":
+                locale = new Locale("pt","BR");
+                languageCode = "pt_BR";
+                languageAvailable = true;
+                break;
+            case "ru":
+                locale = new Locale("ru");
+                languageCode = "ru";
+                languageAvailable = true;
+                break;
+            case "zh":
+                locale = new Locale("zh");
+                languageCode = "zh";
+                languageAvailable = true;
+                break;
+            case "zh-rTW":
+                locale = new Locale("zh","TW");
+                languageCode = "zh_TW";
+                languageAvailable = true;
```

Comment:
_where are the in and iw translations from? They both don't seem to be in our transifex. All languages in the app have to be in transifex, so that the translations can be managed through that._

So when I got the translations to work, I tested all of them. I could not get Hebrew & Indonesian to work so I started researching and found people that encountered similar problems. Look at this question at StackOverflow:

http://stackoverflow.com/questions/8393771/android-not-using-finding-my-hebrew-localization#8393844

It looks like Android looks for values-iw & values-in for Hebrew & Indonesian respectively. So I copied and pasted the resources & renamed them. It's worth noting that if you look at the switch statement, the language codes being sent to the site are still `he` & `id`.

_Can this switch statement be simplified?_

If I understand correctly, I think instead of having a large switch block you want some logic to identify what language is coming in so that way in can be reduced. Most languages are simply and have the string as language code & locale, this could be the default case. Then there would specific cases for specific, problematic, languages. I'll work that in.